addons/web/static/src/search/search_model.js

```diff
@@ -939,10 +940,97 @@ export class SearchModel extends EventBus {
+
+        const searchItems = [];
+
+        for (const [definitionRecordId, definitionRecordName, definitions] of result) {
+            for (const definition of definitions) {
+                const item = Object.values(this.searchItems).find(item =>
+                    item.fieldName === searchItem.fieldName
+                    && item.propertyName === definition.name);
+
+                if (item) {
+                    // already in the list, can happen if we unfold the properties field
+                    // open a form view, edit the property and then go back to the search view
+                    // the label of the property might have been changed
+                    item["description"] = `${definition.string} (${definitionRecordName})`;
+                    searchItems.push(item);
+                    continue;
+                }
+
+                const operator = ["many2many", "tags"].includes(definition.type) ? "in" : undefined;
+
+                this.searchItems[this.nextId] = {
+                    "isActive": false,
+                    "type": "field_property",
+                    "fieldName": searchItem.fieldName,
+                    "propertyName": definition.name,
+                    "propertyDomain": [definitionRecord, "=", definitionRecordId],
+                    "fieldType": "properties",
+                    "description": `${definition.string} (${definitionRecordName})`,
+                    "groupId": this.nextGroupId,
+                    "id": this.nextId,
+                    "autocompleteValues": [],
+                    "definition": definition,
+                    "propertyItemId": searchItem.id,
+                    "relation": definition.comodel,
+                    "selection": definition.selection,
+                    "tags": definition.tags,
+                    "operator": operator,
+                };
+
+                searchItems.push(this.searchItems[this.nextId]);
+
+                this.nextId++;
+                this.nextGroupId++;
+            }
+        }
+
+        return searchItems;
+    }
+
     //--------------------------------------------------------------------------
     // Private methods
     //--------------------------------------------------------------------------
 
+    /**
+     * Fetch the properties definitions.
+     *
+     * @param {string} definitionRecordModel
+     * @param {string} definitionRecordField
+     */
+    async _fetchPropertiesDefinition(definitionRecordModel, definitionRecordField) {
+        let domain = [];
+        if (this.context.active_id) {
+            // assume the active id is the definition record
+            // and show only its properties
+            domain = [['id', '=', this.context.active_id]];
+        }
+
+        const result = await this.orm.webSearchRead(
+            definitionRecordModel,
+            domain,
+            ["display_name", definitionRecordField]
+        );
+
+        return result.records.map(values => [
```

Comment:
Either return a structured object and not a tuple-like with 3 keys, either correctly document the method to explain the return-type.

(Or both :eyes: )